Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,184 @@
+            engineInfo = engineInfoResult.GetValue();
+        }
+        else
+        {
+            QString error = tr("Failed to get engine info.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        QTextStream logStream(&logFile);
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            logFile.close();
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        UpdateProgress(tr("Setting Up Environment"));
+
+        auto currentEnvironmentRequest = ProjectUtils::SetupCommandLineProcessEnvironment();
+        if (!currentEnvironmentRequest.IsSuccess())
+        {
+            QStringToAZTracePrint(currentEnvironmentRequest.GetError());
+            return AZ::Failure(currentEnvironmentRequest.GetError());
+        }
+
+        
+
+        m_exportProjectProcess = new QProcess(this);
+        m_exportProjectProcess->setProcessChannelMode(QProcess::MergedChannels);
+        m_exportProjectProcess->setWorkingDirectory(m_projectInfo.m_path);
+
+        
+        QStringList exportArgs = {"export-project","--project-path", m_projectInfo.m_path};
+        QString commandProgram = QDir(engineInfo.m_path).filePath(GetO3DECLIString());
+
+        m_exportProjectProcess->start(commandProgram, exportArgs);
+        if (!m_exportProjectProcess->waitForStarted())
+        {
+            QString error = tr("Exporting project failed to start.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        while (m_exportProjectProcess->waitForReadyRead(MaxExportTimeMSecs))
+        {
+            QString exportOutput = m_exportProjectProcess->readAllStandardOutput();
+
+            logStream << exportOutput;
+            logStream.flush();
+
+            // Show last line of output
+            if (QStringList strs = exportOutput.split('\n', Qt::SkipEmptyParts);
+                !strs.empty())
+            {
+                UpdateProgress(strs.last());
+            }
+
+            if (QThread::currentThread()->isInterruptionRequested())
+            {
+                // QProcess is unable to kill its child processes so we need to ask the operating system to do that for us
+                auto killProcessArgumentsResult = ConstructKillProcessCommandArguments(QString::number(m_exportProjectProcess->processId()));
+                if (!killProcessArgumentsResult.IsSuccess())
+                {
+                    return AZ::Failure(killProcessArgumentsResult.GetError());
+                }
+                auto killProcessArguments = killProcessArgumentsResult.GetValue();
+
+
+                QProcess killExportProcess;
+
+
+                killExportProcess.setProcessChannelMode(QProcess::MergedChannels);
+                killExportProcess.start(killProcessArguments.front(), killProcessArguments.mid(1));
+                killExportProcess.waitForFinished();
```

Comment:
This is not checking the return value. Its possible that the kill process timed out, so the process is still running. If so, we need to handle that.